include/swift/AST/Types.h

```diff
@@ -336,16 +336,22 @@ class alignas(1 << TypeAlignInBits) TypeBase {
     NumProtocols : 16
   );
 
-  SWIFT_INLINE_BITFIELD_FULL(TypeVariableType, TypeBase, 20+4+20,
-    /// The unique number assigned to this type variable.
-    ID : 20,
-
+  enum { NumTypeVariableTypeOptionBits = 4 };
+  enum { NumTypeVariableTypeIDBits = 64 - NumTypeBaseBits
+                                        - NumTypeVariableTypeOptionBits };
+  static_assert(NumTypeVariableTypeIDBits >= 32,
+                "TypeVariableType bitfield needs refactoring");
+  SWIFT_INLINE_BITFIELD_FULL(TypeVariableType, TypeBase,
+                        NumTypeVariableTypeOptionBits+NumTypeVariableTypeIDBits,
     /// Type variable options.
-    Options : 4,
+    Options : NumTypeVariableTypeOptionBits,
+
+    /// The unique number assigned to this type variable.
+    ID : NumTypeVariableTypeIDBits / 2 + NumTypeVariableTypeIDBits % 2,
 
     ///  Index into the list of type variables, as used by the
     ///  constraint graph.
-    GraphIndex : 20
+    GraphIndex : NumTypeVariableTypeIDBits / 2
```

Comment:
`GraphIndex` has to be the same size as `ID` because it's based on the number of time variables in the constraint system, otherwise  it would lead to the same situation as described in https://github.com/apple/swift/pull/25666 (if that was an overflow check in `Implementation::setGraphIndex`). Better refactoring here would be to investigate elimination of `GraphIndex` and use `ID` in its place.

Reply:
If I make `ID` and `GraphIndex` be automatically the same size, will that be sufficient to land this pull request?

Reply:
I think we can get it as low as 18 bits for each but that would be pushing it, I think 20 bits is in the safe zone. People do have some pretty big literal arrays.

Reply:
It would be great to add overflow check to `Implementation::setGraphIndex` too

Reply:
I've refactored the code to always be 32 bits.